File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    answer = 0
    
    ham=''
    hamlist=[]
    for i in ingredient:
        hamlist.append(i)
        if len(hamlist)>=4:
            if hamlist[-4:]==[1,2,3,1]:
                hamlist.pop()
                hamlist.pop()
                hamlist.pop()
                hamlist.pop()
                answer+=1
            
            
    return answer

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        
        List<Integer> list=new ArrayList<>();
        for(int i:ingredient){
            list.add(i);
           
            int len=list.size();
          
            if(len>3){
                if(list.get(len-1)==1 && 
                  list.get(len-2)==3&&
                  list.get(len-3)==2&&
                  list.get(len-4)==1){
                    answer++;
                for(int r=len-1;r>=len-4;r--) list.remove(r);
                }
            }
        }
        
        return answer;
    }
}